src/lib/comms/sol-mavlink.c

```diff
@@ -703,6 +702,8 @@ sol_mavlink_connect(const char *addr, const struct sol_mavlink_config *config, c
     }
 #endif
 
+    SOL_NULL_CHECK(config->handlers, NULL);
```

Comment:
it doesn't seem right because there is a check for handlers->api_version a few lines above (line 697).

You should check if handlers are null before dereferencing it.